src/RuleList.js

```diff
@@ -136,13 +129,14 @@ export default class RuleList {
    */
   update(name?: string | Object, data?: Object): void {
     const {jss: {plugins}, sheet} = this.options
-    if (typeof name === 'string') {
+    if (typeof name === 'string' && data != null) {
       plugins.onUpdate(data, this.get(name), sheet)
-      return
     }
 
-    for (let index = 0; index < this.index.length; index++) {
-      plugins.onUpdate(name, this.index[index], sheet)
+    if (typeof name === 'object' || name === undefined) {
```

Comment:
@kof I added explicit check here. You can try to test it. I don't know code base and not sure what is the problem here.